src/main/java/org/jooq/lambda/Seq.java

```diff
@@ -9688,15 +9688,8 @@ public T next() {
      * </pre></code>
      */
     static <T> Tuple2<Seq<T>, Seq<T>> splitAt(Stream<? extends T> stream, long position) {
-        return seq(stream)
-            .zipWithIndex()
-            .partition(t -> t.v2 < position)
-            // Explicit type parameters to work around this Eclipse compiler bug:
-            // https://bugs.eclipse.org/bugs/show_bug.cgi?id=455945
-            .map((v1, v2) -> Tuple.<Seq<T>, Seq<T>>tuple(
-                v1.map(t -> t.v1),
-                v2.map(t -> t.v1)
-            ));
+        SeqBuffer<T> buffer = SeqBuffer.of(stream);
+        return tuple(buffer.seq().limit(position), buffer.seq().skip(position));
```

Comment:
@billoneil If I understand it right I think @lukaseder distinguishes between two things:
- single `Seq` is sequential so as a consequence it's not thread safe and two threads cannot operate on the same `Seq`
- there seems to be no restriction to consuming one `Seq` on one thread and another `Seq` on the other thread even if they both "derive" from the same `Seq` (hence need for thread safety in such cases)